ports/mimxrt/machine_timer.c

```diff
@@ -148,6 +148,12 @@ STATIC mp_obj_t machine_timer_make_new(const mp_obj_type_t *type, size_t n_args,
         mp_raise_ValueError(MP_ERROR_TEXT("Timer does not exist"));
     }
 
+    // check, if a timer exists at that channel and stop it first
+    if (MP_STATE_PORT(timer_table)[id] != NULL) {
+        PIT_StopTimer(PIT, channel_no[id]);
+        MP_STATE_PORT(timer_table)[id] = NULL;
+    }
```

Comment:
I'd suggest to just re-use the object:
```c
if (MP_STATE_PORT(timer_table)[id] == NULL) {
    self = m_new_obj_with_finaliser(machine_timer_obj_t);
    MP_STATE_PORT(timer_table)[id] = self;
} else {
    self = MP_STATE_PORT(timer_table)[id];
}
```
That way the hardware timer is mapped 1-1 to an object, and you can retrieve the existing object via `machine.Timer(id)`.  This is how other ports work.